packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -363,8 +390,26 @@ class CreateCommand extends CreateBase {
           printStatusWhenWriting: !creatingNewProject,
           projectType: template,
         );
+        pubContext = PubContext.createPlugin;
         break;
     }
+
+    if (boolArgDeprecated('pub')) {
```

Comment:
These commands are hard to read--can you explain why you're adding this here?

Reply:
Oh, did this used to be called earlier in the _generateModule() and such helper functions?

Reply:
Yes this is moved from the helper-functions.